src/Explain.h

```diff
@@ -32,199 +36,206 @@
-                    continue;
-                }
-                printPrompt("Depth is now " + std::to_string(depthLimit) + "\n");
-            } else if (command[0] == "explain") {
-                std::pair<std::string, std::vector<std::string>> query;
-                if (command.size() == 2) {
-                    query = parseTuple(command[1]);
-                } else {
-                    printStr("Usage: explain relation_name(\"<string element1>\", <number element2>, ...)\n");
-                    continue;
-                }
-                std::unique_ptr<TreeNode> t = prov.explain(query.first, query.second, depthLimit);
-                printTree(std::move(t));
-            } else if (command[0] == "subproof") {
-                std::pair<std::string, std::vector<std::string>> query;
-                int label = -1;
-                if (command.size() > 1) {
-                    query = parseTuple(command[1]);
-                    label = std::stoi(query.second[0]);
-                } else {
-                    printStr("Usage: subproof relation_name(<label>)\n");
-                    continue;
-                }
-                std::unique_ptr<TreeNode> t = prov.explainSubproof(query.first, label, depthLimit);
-                printTree(std::move(t));
-            } else if (command[0] == "explainnegation") {
-                std::pair<std::string, std::vector<std::string>> query;
-                if (command.size() == 2) {
-                    query = parseTuple(command[1]);
-                } else {
-                    printStr(
-                            "Usage: explainnegation relation_name(\"<string element1>\", <number element2>, "
-                            "...)\n");
-                    continue;
-                }
+        if (command.empty()) {
+            return true;
+        }
 
-                size_t i = 1;
-                std::string rules;
-                for (auto rule : prov.getRules(query.first)) {
-                    rules += std::to_string(i) + ": ";
-                    rules += rule;
-                    rules += "\n\n";
-                    i++;
-                }
-                printStr(rules);
+        if (command[0] == "setdepth") {
+            if (command.size() != 2) {
+                printError("Usage: setdepth <depth>\n");
+                return true;
+            }
+            try {
+                ExplainConfig::getExplainConfig().depthLimit = std::stoi(command[1]);
+            } catch (std::exception& e) {
+                printError("<" + command[1] + "> is not a valid depth\n");
+                return true;
+            }
+            printInfo("Depth is now " + std::to_string(ExplainConfig::getExplainConfig().depthLimit) + "\n");
+        } else if (command[0] == "explain") {
+            std::pair<std::string, std::vector<std::string>> query;
+            if (command.size() == 2) {
+                query = parseTuple(command[1]);
+            } else {
```

Comment:
This would be better if you started the "explain" section with `if (command.size !=2) { blah blah return true }` to handle the exceptional case, then the rest is the normal case.